### PR TITLE
Adds pid_mode feature to docker driver

### PR DESCRIPTION
**Pull request by angelbarrera92**
Solves #1534 
Adds `pid_mode` configuration parameter listen in [docker ansible module](https://docs.ansible.com/ansible/latest/modules/docker_container_module.html)

Thanks!

### CODE REVIEW

File: test/unit/model/v2/test_platforms_section.py [hunk 1]
```diff
@@ -50,6 +50,8 @@ def _model_platforms_docker_section_data():
             },
             'command':
             'sleep infinity',
+            'pid_mode':
```
Comment (lwm) [inline test/unit/model/v2/test_platforms_section.py]: `%s/:/,/`

Also, why was host added here? Should it not only be `pid_mode`?

Comment (lwm) [reply test/unit/model/v2/test_platforms_section.py]: Oh wait, read it incorrectly! All good.

**Comment by lwm** [comment]
Can you rebase @angelbarrera92?

Added to agenda in https://github.com/ansible/community/issues/427. Please join if you have time!

**Comment by angelbarrera92** [comment]
@lwm i  will rebase asap ;)

**Comment by angelbarrera92** [comment]
Could you re-launch the tests of this change? I think it failed by timeout:

`The job exceeded the maximum time limit for jobs, and has been terminated.`

Thanks!

**Comment by decentral1se** [comment]
Restarted @angelbarrera92! Thanks for your patience :+1:

**Comment by webknjaz** [comment]
Hey @angelbarrera92, it looks like there are some conflicts with the current `master`. Would you mind resolving them, please?

**Comment by angelbarrera92** [comment]
Resolved!

**Comment by decentral1se** [comment]
:rocket:

**Pull request merged**